contracts/CurrencyNetwork.sol

```diff
@@ -393,20 +435,25 @@ contract CurrencyNetwork is CurrencyNetworkInterface, Ownable, Authorizable, Des
     function _directTransfer(
         address _sender,
         address _receiver,
-        uint32 _value
+        uint128 _value
     )
         internal
-        returns (uint32 fees)
+        returns (uint128 fees)
     {
         Account memory account = _loadAccount(_sender, _receiver);
 
-        uint32 creditlineReceived = account.creditlineReceived;
-        fees = _calculateFees(_value, account.balance, capacityImbalanceFeeDivisor);
-        int64 newBalance = account.balance - _value - fees;
+        uint32 mtime = account.mtime;
+        int136 interests = _calculateInterests(account.balance, mtime, account.interestRateGiven, account.interestRateReceived);
```

Comment:
What about it?
Overflows are checked a few lines below, it could save gas in case of failure due to overflows of the interests to check it also earlier if that's what you imply (but I believe it would be extremely unfortunate).